_posts/2018-08-30-on-the-fly-transformations.md

```diff
@@ -0,0 +1,283 @@
+This is the case of our system. Without any modifications, when we look at the trajectory of our system, things will become very ugly, very fast:
+
+
+```python
+import warnings
+warnings.filterwarnings('ignore') # some attributes are missing 
+import MDAnalysis as mda
+import nglview as nv
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+nv.show_mdanalysis(u)
+```
+
+
+    NGLWidget(count=11)
+
+
+Using `trjconv`, one way to may every molecules whole again would be:
+
+    gmx trjconv -f pept_in_memb.xtc -s pept_in_memb.tpr -pbc mol -o output.xtc
+    
+In MDAnalysis this can be done with the `unwrap` transformation, which takes an AtomGroup as argument. This can be done as follows:
+
+
+```python
+# a custom atom group can be passed as an argument. In this case we will use all the atoms
+# in the Universe u
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+ag = u.atoms
+# we define the transformation
+transformation = mda.transformations.unwrap(ag)
+# we add the transformation to the trajectory in it will be applied everytime we read a frame
+u.trajectory.add_transformations(transformation)
+# we can do other things with the trajectory without having to generate a file with the transformed
+# trajectory
+# for ts in u.trajectory:
+#     analysis1()
+#     analysis2()
+nv.show_mdanalysis(u)
+```
+
+
+    NGLWidget(count=11)
+
+
+as you can see, the artifacts caused by the atoms crossing the boundaries of the unit cell are now gone.
+
+### Example 2: what if we also want to center the peptide in the unit cell?
+In that case, using `trjconv` we would do something like this:
+   
+    gmx trjconv -f pept_in_memb.xtc -s pept_in_memb.tpr -pbc mol -center -o output.xtc
+
+And we choose `Protein` as the group to be centered.
+
+In MDAnalysis we use the `center_in_box` transformation. This function takes an AtomGroup as a mandatory argument. Optional arguments include `weights`, which is used to calculate the weighted center of the given AtomGroup (if weights='mass' then the center of mass is calculated), `center_to` which is used when the user needs to center the AtomGroup in a custom point instead of the center of the unit cell, and `wrap` which, if `True`, causes all the atoms of the AtomGroup to be moved to the unit cell before calculating the weighted center.
+This transformation can be performed as follows:
+
+
+```python
+u = mda.Universe('pept_in_memb.tpr', 'pept_in_memb.xtc')
+prot = u.select_atoms("protein")
+ag = u.atoms
+# we will use mass as weights for the center calculation
+transformations = (mda.transformations.unwrap(ag),
+                   mda.transformations.center_in_box(prot, weights='mass'),
+                   mda.transformations.wrap(ag, compound='fragments'))
+u.trajectory.add_transformations(*transformations)
+nv.show_mdanalysis(u)
+```
+
+
+    NGLWidget(count=11)
+
+
+You can see that the `transformations` workflow above has three steps:
```

Comment:
explain this before. Say the transformation now consists of three steps and explain them like you do here. Then show the code. The code doesn't come as a surprise then.